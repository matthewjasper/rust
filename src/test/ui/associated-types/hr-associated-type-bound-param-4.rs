trait X<'a, T>
where
    for<'b> (T,): X<'b, T>,
    for<'b> <(T,) as X<'b, T>>::U: Clone,
{
    type U: ?Sized;
    fn f(x: &<(T,) as X<'_, T>>::U) {
        <<(T,) as X<'_, T>>::U>::clone(x);
    }
}

impl<S, T> X<'_, T> for (S,) {
    type U = str;
    //~^ ERROR overflow evaluating the requirement `T: std::marker::Sized`
    //~| ERROR the trait bound `str: std::clone::Clone` is not satisfied
    //~| ERROR overflow evaluating the requirement `(T,): X<\'_, T>`
}

pub fn main() {
    <(i32,) as X<i32>>::f("abc");
}
