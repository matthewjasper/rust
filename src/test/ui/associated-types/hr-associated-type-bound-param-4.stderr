error[E0275]: overflow evaluating the requirement `T: std::marker::Sized`
  --> $DIR/hr-associated-type-bound-param-4.rs:1:13
   |
LL | trait X<'a, T>
   |             ^
   | 
  ::: $SRC_DIR/libcore/marker.rs:LL:COL
   |
LL | pub trait Sized {
   | --------------- required by this bound in `std::marker::Sized`
   |
   = help: consider adding a `#![recursion_limit="256"]` attribute to your crate (`hr_associated_type_bound_param_4`)
   = note: required because of the requirements on the impl of `X<'_, T>` for `(T,)`
   = note: required because of the requirements on the impl of `X<'_, T>` for `(T,)`
   = note: required because of the requirements on the impl of `X<'_, T>` for `(T,)`
   = note: required because of the requirements on the impl of `X<'_, T>` for `(T,)`
   = note: required because of the requirements on the impl of `X<'_, T>` for `(T,)`
   = note: required because of the requirements on the impl of `X<'_, T>` for `(T,)`
   = note: required because of the requirements on the impl of `X<'_, T>` for `(T,)`
   = note: required because of the requirements on the impl of `X<'_, T>` for `(T,)`
   = note: required because of the requirements on the impl of `X<'_, T>` for `(T,)`
   = note: required because of the requirements on the impl of `X<'_, T>` for `(T,)`
   = note: required because of the requirements on the impl of `X<'_, T>` for `(T,)`
   = note: required because of the requirements on the impl of `X<'_, T>` for `(T,)`
   = note: required because of the requirements on the impl of `X<'_, T>` for `(T,)`
   = note: required because of the requirements on the impl of `X<'_, T>` for `(T,)`
   = note: required because of the requirements on the impl of `X<'_, T>` for `(T,)`
   = note: required because of the requirements on the impl of `X<'_, T>` for `(T,)`
   = note: required because of the requirements on the impl of `X<'_, T>` for `(T,)`
   = note: required because of the requirements on the impl of `X<'_, T>` for `(T,)`
   = note: required because of the requirements on the impl of `X<'_, T>` for `(T,)`
   = note: required because of the requirements on the impl of `X<'_, T>` for `(T,)`
   = note: required because of the requirements on the impl of `X<'_, T>` for `(T,)`
   = note: required because of the requirements on the impl of `X<'_, T>` for `(T,)`
   = note: required because of the requirements on the impl of `X<'_, T>` for `(T,)`
   = note: required because of the requirements on the impl of `X<'_, T>` for `(T,)`
   = note: required because of the requirements on the impl of `X<'_, T>` for `(T,)`
   = note: required because of the requirements on the impl of `X<'_, T>` for `(T,)`
   = note: required because of the requirements on the impl of `X<'_, T>` for `(T,)`
   = note: required because of the requirements on the impl of `X<'_, T>` for `(T,)`
   = note: required because of the requirements on the impl of `X<'_, T>` for `(T,)`
   = note: required because of the requirements on the impl of `X<'_, T>` for `(T,)`
   = note: required because of the requirements on the impl of `X<'_, T>` for `(T,)`
   = note: required because of the requirements on the impl of `X<'_, T>` for `(T,)`
   = note: required because of the requirements on the impl of `X<'_, T>` for `(T,)`
   = note: required because of the requirements on the impl of `X<'_, T>` for `(T,)`
   = note: required because of the requirements on the impl of `X<'_, T>` for `(T,)`
   = note: required because of the requirements on the impl of `X<'_, T>` for `(T,)`
   = note: required because of the requirements on the impl of `X<'_, T>` for `(T,)`
   = note: required because of the requirements on the impl of `X<'_, T>` for `(T,)`
   = note: required because of the requirements on the impl of `X<'_, T>` for `(T,)`
   = note: required because of the requirements on the impl of `X<'_, T>` for `(T,)`
   = note: required because of the requirements on the impl of `X<'_, T>` for `(T,)`
   = note: required because of the requirements on the impl of `X<'_, T>` for `(T,)`
   = note: required because of the requirements on the impl of `X<'_, T>` for `(T,)`
   = note: required because of the requirements on the impl of `X<'_, T>` for `(T,)`
   = note: required because of the requirements on the impl of `X<'_, T>` for `(T,)`
   = note: required because of the requirements on the impl of `X<'_, T>` for `(T,)`
   = note: required because of the requirements on the impl of `X<'_, T>` for `(T,)`
   = note: required because of the requirements on the impl of `X<'_, T>` for `(T,)`
   = note: required because of the requirements on the impl of `X<'_, T>` for `(T,)`
   = note: required because of the requirements on the impl of `X<'_, T>` for `(T,)`
   = note: required because of the requirements on the impl of `X<'_, T>` for `(T,)`
   = note: required because of the requirements on the impl of `X<'_, T>` for `(T,)`
   = note: required because of the requirements on the impl of `X<'_, T>` for `(T,)`
   = note: required because of the requirements on the impl of `X<'_, T>` for `(T,)`
   = note: required because of the requirements on the impl of `X<'_, T>` for `(T,)`
   = note: required because of the requirements on the impl of `X<'_, T>` for `(T,)`
   = note: required because of the requirements on the impl of `X<'_, T>` for `(T,)`
   = note: required because of the requirements on the impl of `X<'_, T>` for `(T,)`
   = note: required because of the requirements on the impl of `X<'_, T>` for `(T,)`
   = note: required because of the requirements on the impl of `X<'_, T>` for `(T,)`
   = note: required because of the requirements on the impl of `X<'_, T>` for `(T,)`
   = note: required because of the requirements on the impl of `X<'_, T>` for `(T,)`
   = note: required because of the requirements on the impl of `X<'_, T>` for `(T,)`
   = note: required because of the requirements on the impl of `X<'_, T>` for `(T,)`

error[E0277]: the trait bound `str: std::clone::Clone` is not satisfied
  --> $DIR/hr-associated-type-bound-param-4.rs:13:14
   |
LL | trait X<'a, T>
   |       - required by a bound in this
...
LL |     for<'b> <(T,) as X<'b, T>>::U: Clone,
   |                                    ----- required by this bound in `X`
...
LL |     type U = str;
   |              ^^^ the trait `std::clone::Clone` is not implemented for `str`

error[E0275]: overflow evaluating the requirement `T: std::marker::Sized`
  --> $DIR/hr-associated-type-bound-param-4.rs:1:13
   |
LL | trait X<'a, T>
   |             ^
   | 
  ::: $SRC_DIR/libcore/marker.rs:LL:COL
   |
LL | pub trait Sized {
   | --------------- required by this bound in `std::marker::Sized`
   |
   = help: consider adding a `#![recursion_limit="256"]` attribute to your crate (`hr_associated_type_bound_param_4`)
   = note: required because of the requirements on the impl of `X<'_, T>` for `(T,)`
   = note: required because of the requirements on the impl of `X<'_, T>` for `(T,)`
   = note: required because of the requirements on the impl of `X<'_, T>` for `(T,)`
   = note: required because of the requirements on the impl of `X<'_, T>` for `(T,)`
   = note: required because of the requirements on the impl of `X<'_, T>` for `(T,)`
   = note: required because of the requirements on the impl of `X<'_, T>` for `(T,)`
   = note: required because of the requirements on the impl of `X<'_, T>` for `(T,)`
   = note: required because of the requirements on the impl of `X<'_, T>` for `(T,)`
   = note: required because of the requirements on the impl of `X<'_, T>` for `(T,)`
   = note: required because of the requirements on the impl of `X<'_, T>` for `(T,)`
   = note: required because of the requirements on the impl of `X<'_, T>` for `(T,)`
   = note: required because of the requirements on the impl of `X<'_, T>` for `(T,)`
   = note: required because of the requirements on the impl of `X<'_, T>` for `(T,)`
   = note: required because of the requirements on the impl of `X<'_, T>` for `(T,)`
   = note: required because of the requirements on the impl of `X<'_, T>` for `(T,)`
   = note: required because of the requirements on the impl of `X<'_, T>` for `(T,)`
   = note: required because of the requirements on the impl of `X<'_, T>` for `(T,)`
   = note: required because of the requirements on the impl of `X<'_, T>` for `(T,)`
   = note: required because of the requirements on the impl of `X<'_, T>` for `(T,)`
   = note: required because of the requirements on the impl of `X<'_, T>` for `(T,)`
   = note: required because of the requirements on the impl of `X<'_, T>` for `(T,)`
   = note: required because of the requirements on the impl of `X<'_, T>` for `(T,)`
   = note: required because of the requirements on the impl of `X<'_, T>` for `(T,)`
   = note: required because of the requirements on the impl of `X<'_, T>` for `(T,)`
   = note: required because of the requirements on the impl of `X<'_, T>` for `(T,)`
   = note: required because of the requirements on the impl of `X<'_, T>` for `(T,)`
   = note: required because of the requirements on the impl of `X<'_, T>` for `(T,)`
   = note: required because of the requirements on the impl of `X<'_, T>` for `(T,)`
   = note: required because of the requirements on the impl of `X<'_, T>` for `(T,)`
   = note: required because of the requirements on the impl of `X<'_, T>` for `(T,)`
   = note: required because of the requirements on the impl of `X<'_, T>` for `(T,)`
   = note: required because of the requirements on the impl of `X<'_, T>` for `(T,)`
   = note: required because of the requirements on the impl of `X<'_, T>` for `(T,)`
   = note: required because of the requirements on the impl of `X<'_, T>` for `(T,)`
   = note: required because of the requirements on the impl of `X<'_, T>` for `(T,)`
   = note: required because of the requirements on the impl of `X<'_, T>` for `(T,)`
   = note: required because of the requirements on the impl of `X<'_, T>` for `(T,)`
   = note: required because of the requirements on the impl of `X<'_, T>` for `(T,)`
   = note: required because of the requirements on the impl of `X<'_, T>` for `(T,)`
   = note: required because of the requirements on the impl of `X<'_, T>` for `(T,)`
   = note: required because of the requirements on the impl of `X<'_, T>` for `(T,)`
   = note: required because of the requirements on the impl of `X<'_, T>` for `(T,)`
   = note: required because of the requirements on the impl of `X<'_, T>` for `(T,)`
   = note: required because of the requirements on the impl of `X<'_, T>` for `(T,)`
   = note: required because of the requirements on the impl of `X<'_, T>` for `(T,)`
   = note: required because of the requirements on the impl of `X<'_, T>` for `(T,)`
   = note: required because of the requirements on the impl of `X<'_, T>` for `(T,)`
   = note: required because of the requirements on the impl of `X<'_, T>` for `(T,)`
   = note: required because of the requirements on the impl of `X<'_, T>` for `(T,)`
   = note: required because of the requirements on the impl of `X<'_, T>` for `(T,)`
   = note: required because of the requirements on the impl of `X<'_, T>` for `(T,)`
   = note: required because of the requirements on the impl of `X<'_, T>` for `(T,)`
   = note: required because of the requirements on the impl of `X<'_, T>` for `(T,)`
   = note: required because of the requirements on the impl of `X<'_, T>` for `(T,)`
   = note: required because of the requirements on the impl of `X<'_, T>` for `(T,)`
   = note: required because of the requirements on the impl of `X<'_, T>` for `(T,)`
   = note: required because of the requirements on the impl of `X<'_, T>` for `(T,)`
   = note: required because of the requirements on the impl of `X<'_, T>` for `(T,)`
   = note: required because of the requirements on the impl of `X<'_, T>` for `(T,)`
   = note: required because of the requirements on the impl of `X<'_, T>` for `(T,)`
   = note: required because of the requirements on the impl of `X<'_, T>` for `(T,)`
   = note: required because of the requirements on the impl of `X<'_, T>` for `(T,)`
   = note: required because of the requirements on the impl of `X<'_, T>` for `(T,)`
   = note: required because of the requirements on the impl of `X<'_, T>` for `(T,)`

error: aborting due to 3 previous errors

Some errors have detailed explanations: E0275, E0277.
For more information about an error, try `rustc --explain E0275`.
